--- Command_Lists.c	2015-03-20 17:40:18.288258800 +0900
+++ Command_Lists.c	2018-12-21 16:00:26.074704200 +0900
@@ -29,6 +29,7 @@
 #include "throughput.h"
 #include "atheros_stack_offload.h"
 #include "shell.h"
+#include "tio_demo.h"
 
 #if ENABLE_STACK_OFFLOAD
 #if ENABLE_HTTPS_SERVER || ENABLE_HTTPS_CLIENT
@@ -83,6 +84,7 @@
 #endif
 #endif
 #endif
+   { "tio", tio_main},
    { "?",         Shell_command_list },     
    { NULL,        NULL } 
 };
--- Tasks.c  2015-04-01 18:48:27.821462100 +0900
+++ Tasks.c  2018-12-21 16:05:16.605868200 +0900
@@ -18,6 +18,7 @@
 
 #include "main.h"
 #include "throughput.h"
+#include "tio_task_impl.h"
 
 /*
 ** MQX initialization information
@@ -29,8 +30,10 @@
    { WMICONFIG_TASK1,  wmiconfig_Task1,  3000,    9,      "WMICONFIG1",	MQX_AUTO_START_TASK,  0,      0           },   
    { WMICONFIG_TASK2,  wmiconfig_Task2,  3000,    9,      "WMICONFIG2", MQX_AUTO_START_TASK,  0,      0           },
 #if DEMOCFG_ENABLE_SERIAL_SHELL
-    { SHELL_TASK,   Shell_Task,        2000,   12,      "Shell",    MQX_AUTO_START_TASK,      0,      0           },
+    { SHELL_TASK,   Shell_Task,        4000,   12,      "Shell",    MQX_AUTO_START_TASK,      0,      0           },
 #endif
+   { TIO_MQTT_TASK,    _tio_task,        4000,    9,      "kii_mqtt_task",    0,  0,      0           },
+   { TIO_UPDATER_TASK, _tio_task,        4000,    9,      "task_update_state",    0,  0,      0           },
 
     {0}
 };
--- throughput.h	2015-04-01 18:48:27.830462600 +0900
+++ throughput.h	2019-01-08 16:37:01.645176000 +0900
@@ -32,11 +32,11 @@
 #define MAX_END_MARK_RETRY                50   // Max number of End Mark retries in UDP Tx test
 #define MSEC_HEARTBEAT 			  10000
 #define MAX_TASKS_ALLOWED 		  2	 // For now, we allow ony 2 tasks
-#define MAX_STREAMS_ALLOWED 		  2	 // Max number of allowed TCP/UDP streams- limited by memory
+#define MAX_STREAMS_ALLOWED 		  3	 // Max number of allowed TCP/UDP streams- limited by memory
 #define TCP_CONNECTION_WAIT_TIME          50 
 #define UDP_CONNECTION_WAIT_TIME          500
 #define UDP_CONNECTION_WAIT_TIME_MULTI_SOCK 50
